Update Program.cs

tavisca-jkanathe opened the pull request:
@sirius17 @dagarwal 
solution to program 1 of  assignment 2.

sirius17 commented on Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs:
```diff
@@ -14,16 +14,119 @@ static void Main(string[] args)
+			{
+				int B = int.Parse(nums[1]);
+				int C = int.Parse(nums[2]);
+				if(C%B!=0)
```
Good catch on ensuring that the integer division is without any remainder.

tavisca-jkanathe commented:
Sure, i will make the required changes
----- Original Message -----
From: Nikhil Prasad <notifications@github.com>
To: tavisca-bootcamp/language-basics-01 <language-basics-01@noreply.github.com>
Cc: tavisca-jkanathe <jkanathe@tavisca.in>, Author <author@noreply.github.com>
Sent: Tue, 25 Jun 2019 10:43:09 +0530 (IST)
Subject: Re: [tavisca-bootcamp/language-basics-01] Update Program.cs (#49)

sirius17 requested changes on this pull request.

Please make the requested changes.

> +					int find = nums[2].IndexOf('?');
+						string oldC = C.ToString();
+						char result = oldC[find];
+						String newC = nums[2].Replace('?',result);
+						if(string.Equals(newC,oldC))
+						{
+							return int.Parse(result.ToString());
+						}
+                    else
+						return -1;
+				}
+				
+			}
+			else
+				return -1;
+            throw new NotImplementedException();

This line can be removed since this method now has an implementation :)


> +				int C = int.Parse(nums[2]);
+				if(C%B!=0)
+				{
+					return -1;
+				}
+				else
+				{
+					
+					int A=C/B;
+					if(A.ToString().Length!= nums[0].Length)
+					{
+						return -1;
+					}
+					else
+					{
+						int find = nums[0].IndexOf('?');

This entire implementation seems to be repeating quite a bit in the code. It is a good candidate for refactoring into a separate method.

> +            char[]  splitchars = new char[2]{'*','='};
+            String[] nums=equation.Split(splitchars);
+            int check=-1;
+            for(int i=0;i<=2;i++)  
+            {
+                if(nums[i].Contains("?"))
+                {
+                    check=i;
+					break;
+                }
+			}
+			if(check==0)
+			{
+				int B = int.Parse(nums[1]);
+				int C = int.Parse(nums[2]);
+				if(C%B!=0)

Good catch on ensuring that the integer division is without any remainder.

-- 
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub:
https://github.com/tavisca-bootcamp/language-basics-01/pull/49#pullrequestreview-253769997

sirius17 commented on Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs:
```diff
@@ -14,16 +14,91 @@ static void Main(string[] args)
-        public static int FindDigit(string equation)
-        {
+	// To find and replace the digit and then to check whether it is valid or not
+	public static int findMissing(int number,int position, String[] nums)
```
Naming conventions in .NET follow PascalCasing. So the function should be named `FindMissing()`.

sirius17 replied:
What is the value of passing `position` and 'nums[]` when you could have simply passed num[position] directly?

sirius17 commented on Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs:
```diff
@@ -14,16 +14,91 @@ static void Main(string[] args)
-            throw new NotImplementedException();
+            char[]  splitchars = new char[2]{'*','='};
+            String[] nums=equation.Split(splitchars);
+            int check=-1;
```
Is there a way you can get rid of this check variable? Driving if else workflows from magic variables and their values makes code hard to read and understand by anyone other than the author.

sirius17 commented on Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs:
```diff
@@ -14,16 +14,91 @@ static void Main(string[] args)
+                if(nums[i].Contains("?"))
+                {
+                    check=i;
+					break;
```
If would also be useful to see if we can remove the break. Linear implementations without too many branches and breaks make for easier reading and maintainability.

tavisca-jkanathe commented:
Thank you for corrections, I will do required changes and communicate to you.
----- Original Message -----
From: Nikhil Prasad <notifications@github.com>
To: tavisca-bootcamp/language-basics-01 <language-basics-01@noreply.github.com>
Cc: tavisca-jkanathe <jkanathe@tavisca.in>, Author <author@noreply.github.com>
Sent: Sun, 30 Jun 2019 20:06:34 +0530 (IST)
Subject: Re: [tavisca-bootcamp/language-basics-01] Update Program.cs (#49)

sirius17 requested changes on this pull request.



>          }
-
-        public static int FindDigit(string equation)
-        {
+	// To find and replace the digit and then to check whether it is valid or not
+	public static int findMissing(int number,int position, String[] nums)

Naming conventions in .NET follow PascalCasing. So the function should be named `FindMissing()`.

>              // Add your code here.
-            throw new NotImplementedException();
+            char[]  splitchars = new char[2]{'*','='};
+            String[] nums=equation.Split(splitchars);
+            int check=-1;

Is there a way you can get rid of this check variable? Driving if else workflows from magic variables and their values makes code hard to read and understand by anyone other than the author.

>              // Add your code here.
-            throw new NotImplementedException();
+            char[]  splitchars = new char[2]{'*','='};
+            String[] nums=equation.Split(splitchars);
+            int check=-1;
+            for(int i=0;i<=2;i++)  
+            {
+                if(nums[i].Contains("?"))
+                {
+                    check=i;
+					break;

If would also be useful to see if we can remove the break. Linear implementations without too many branches and breaks make for easier reading and maintainability.

>          }
-
-        public static int FindDigit(string equation)
-        {
+	// To find and replace the digit and then to check whether it is valid or not
+	public static int findMissing(int number,int position, String[] nums)

What is the value of passing `position` and 'nums[]` when you could have simply passed num[position] directly?

-- 
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub:
https://github.com/tavisca-bootcamp/language-basics-01/pull/49#pullrequestreview-256076818
